lib/cisco_node_utils/interface.rb

```diff
@@ -1988,5 +1988,17 @@ def set_range_based_params(list, param_name)
         end
       end
     end
+
+    def purge_config=(val)
+      fail ArgumentError, 'Interface MUST be ethernet' unless
+        @name[/ethernet/]
+      fail ArgumentError, 'Only true is allowed for this config' unless val
```

Comment:
I would remove the second `fail` and support both `true` and `false`.  You can simply add an early `return unless val` at the beginning of the method.